packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,193 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import { EVENTS } from './utils/const'
+import { setupEvents, removeEvent } from './utils/mapEventHelpers'
+
+const enableControls = (enable = true) => ({
+  draggable: enable,
+  zoomControl: enable,
+  scrollwheel: enable,
+  disableDoubleClickZoom: enable,
+})
+
+@themed(['Gmap_FreeDrawTootip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+    mapControls: PropTypes.object,
+    shapes: PropTypes.object,
+    multipleShapes: PropTypes.bool,
+    disabled: PropTypes.bool,
+    dataStyle: PropTypes.object,
+  }
+
+  static defaultProps = {
+    theme: {},
+    map: {},
+    multipleShapes: false,
+    disabled: false,
+    shapes: {},
+    dataStyle: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      drawing: false,
+    }
+    this.polygons = Object.keys(this.props.shapes).length ? [this.addShapeToGMaps(
```

Comment:
The only time the google maps deals with this format is when taking in new paths. Never when returning paths so it doesn't make sense to accept it

Reply:
I agree with you though @grenti , I don't think we should be manipulating data coming in. It should be in the format Gmaps wants it in so I/O is the same.  It may not be the same data if rent-js used it and then we'd have to rewrite it. We should probably strip it out TBH.

Reply:
@yasabere it actually makes sense what @grenti is saying. You're assuming that the format is always in at listing service format and are then mutating it into what Gmaps wants.  That seems wrong.